docs/ansible/coding-conventions.md

```diff
@@ -0,0 +1,669 @@
+    service:
+      name: critical
+      state: restarted
+  rescue:
+   command: shutdown -h now
+```
+
+## Order in playbook
+
+Playbook definitions should follow this order.
+
+* `name`
+* `hosts`
+* `remote_user`
+* `become`
+* `vars`
+* `pre_tasks`
+* `roles`
+* `tasks`
+
+```yaml
+---
+- name: update root authorized_keys for all machines
+  hosts:
+    - all
+  become: true
+
+  vars:
+    root_keys_users:
+      - user: user1
+        key: |
+          ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIHkCVF05JvfkrfOOESivOxV4N8+A/EMEkF7/nCQMRoQg
+        enabled: true
+
+  pre_tasks:
+    - name: add custom root users
+      set_fact:
+        root_keys_users: "{{ root_keys_users | union( root_keys_users_custom|default([]) ) }}"
+
+  roles:
+    - role: root-keys
+
+  tasks:
+    - name: print out debug message
+      ansible.builtin.debug:
+        msg: "fee foo faa"
+```
+
+ Why do this?
+
+A common order makes playbooks consistent and easier to read for your dear colleagues. Think of them when you write.
+
+## Order in task declaration
+
+A task should be declared in this order.
+
+* `name`
+* module and the arguments for the module
+* arguments for the task in alphabetic order
+* `tags` at the bottom
+
+```yaml
+---
+- name: unhold packages
+  ansible.builtin.shell: 'apt-mark -s unhold {{ item }} && apt-mark unhold {{ item }}'
+  args:
+    executable: /bin/bash
+  changed_when: apt_mark_unhold.rc == 0 and "already" not in apt_mark_unhold.stdout
+  failed_when: false
+  loop: '{{ packages_ubuntu_unhold }}'
+  register: apt_mark_unhold
+  tags:
+    - apt_unhold
+---
+```
+
+### Why do this?
+
+Reason is the same as "order of playbook". To make tasks more consistent and easier to read. Help your colleagues.
+
+## Air and tabulators
+
+Air, one of the **most important thing** for humans and **for code**! It must be an empty line before `vars`, `pre_tasks`, `roles` and `tasks`, and before each task in the tasks definition. Tabulator stops must be set to two, `2`, spaces.
+
+### Why do this?
+
+This creates a pretty and tidy code which is easy to read, both for dyslectic and non dyslectic people. There is no excuse not to do this.
+
+## Variable names
+
+Always use `snake_case` for variable names.
+
+✅ ***Good***
+
+```yaml
+---
+- name: set my variables
+  ansible.builtin.set_fact:
+    a_boolean: false
+    an_int: 101
+    a_string: bar
+```
+❌ ***Bad***
+
+```yaml
+---
+- name: set my variables
+  ansible.builtin.set_fact:
+    aBoolean: false
+    anint: 101
+    A_STRING: bar
+```
+ Why do this?
+
+Ansible already uses `snake_case` for variables in it's examples. Consistent naming of variables keeps the code tidy and gives better readability.
+
+## Content Organization
+
+Your usage of Ansible should fit your needs, however, not ours, so feel free to modify this approach and organize as you see fit.
+
+Always try to use Ansible’s “roles” to organize your playbook content  organization feature.
+
+## Directory Layout
```

Comment:
just add link to official OrchidE docs about file structure,

add some minimal example here. (the example below is not accurate with the plugin right now)